src/main/java/org/apache/pulsar/ecosystem/io/lakehouse/sink/PrimitiveFactory.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.ecosystem.io.lakehouse.sink;
+
+import java.nio.ByteBuffer;
+import org.apache.avro.Schema;
+import org.apache.commons.lang.StringUtils;
+import org.apache.pulsar.common.schema.SchemaType;
+
+/**
+ *  PrimitiveFactory provides a way to get different PulsarObject according to the given schema type.
+ */
+public class PrimitiveFactory {
+
+    public static PulsarObject getPulsarPrimitiveObject(SchemaType schemaType, Object value,
+                                                         String overrideFieldName) {
+        PulsarObject object;
+        switch (schemaType) {
+            case BYTES:
+                object = new PulsarObject<>(ByteBuffer.wrap((byte[]) value), Schema.create(Schema.Type.BYTES));
+                break;
+            case STRING:
+                object = new PulsarObject<>((String) value, Schema.create(Schema.Type.STRING));
+                break;
+            default:
+                throw new RuntimeException("Failed to build pulsar object, the given type '" + schemaType + "' "
```

Comment:
We need to support all primitive type that Pulsar supported.

Reply:
I'm not sure if it needs to support INT, BOOLEAN or DOUBLE. Because it looks useless to save an int or boolean into the table. so I want to support it when there has someone who wants it.